### PR TITLE
Better label errors/include label response body

### PR DESCRIPTION
**Pull request by laurenleach**
- Better label errors
- Include label response body
- Always return ticket even if label error

### CODE REVIEW

File: pkg/connector/ticket.go [hunk 1]
```diff
@@ -150,27 +152,29 @@ func (s *ServiceNow) CreateTicket(ctx context.Context, ticket *v2.Ticket, schema
...
+	serviceCatalogRequestedItem, updateErr := s.client.UpdateServiceCatalogRequestItem(ctx,
 		serviceCatalogRequestedItem.Id,
 		&servicenow.RequestedItemUpdatePayload{
 			Description: ticket.Description,
 		},
 	)
-	if err != nil {
-		return nil, nil, fmt.Errorf("servicenow-connector: failed to update catalog requested item %s: %w", serviceCatalogRequestedItem.Id, err)
+	if updateErr != nil {
+		updateErr = fmt.Errorf("servicenow-connector: failed to update catalog requested item description: %w", err)
```
Comment (btipling) [inline pkg/connector/ticket.go:164]: did you want this to be `updateErr` at the end?

```go
updateErr = fmt.Errorf("servicenow-connector: failed to update catalog requested item description: %w", updateErr)
```

File: pkg/connector/ticket.go [hunk 2]
```diff
@@ -291,13 +290,20 @@ func (s *ServiceNow) serviceCatalogRequestItemToTicket(ctx context.Context, requ
...
 		CreatedAt:    timestamppb.New(createdAt),
 		UpdatedAt:    timestamppb.New(updatedAt),
 		CompletedAt:  completedAt,
-	}, nil, nil
+	}
+
+	labels, err := s.client.GetLabelsForRequestedItem(ctx, requestedItem.Id)
+	if err != nil {
+		return t, nil, fmt.Errorf("failed to get labels for requested item %s: %w", requestedItem.Id, err)
+	}
```
Comment (btipling) [inline pkg/connector/ticket.go:303]: What affect does this ordering change have?

Comment (laurenleach) [reply pkg/connector/ticket.go:303]: we may have successfully fetched the ticket, but not the labels, so this still returns the ticket even if we couldn't get the labels

I'm not sure if we really need to return the label error, we don't even use the labels

File: pkg/connector/ticket.go (hunk 1 above)
Comment (laurenleach) [reply pkg/connector/ticket.go:164]: yes, thanks

**Review by btipling — APPROVED** [review]

**Pull request merged**